File: Tests/server/server_executors/TextGenerator.cpp
#include "TextGenerator.h"

static constexpr size_t smallStringSize = 15;

TextGenerator::TextGenerator(std::string_view data) :
	data(data),
	offset(0)
{

}

std::string_view TextGenerator::generate()
{
	if (offset >= data.size())
	{
		return {};
	}

	lastResult = data.substr(offset, (std::min)(smallStringSize, data.size() - offset));

	offset += lastResult.size();

	return lastResult;
}

File: Tests/executors/Python/core/TextGenerator.py
from web_framework_api import *


class TextGenerator(ChunkGenerator):
    def __init__(self, data: str):
        self._data = data
        self._offset = 0
        self._last_result = ""
        self._STRING_SIZE = 15

    def generate(self):
        if self._offset > len(self._data):
            return ""

        self._last_result = self._data[
            self._offset:self._offset + min(self._STRING_SIZE, len(self._data) - self._offset)]

        self._offset += len(self._last_result)

        return self._last_result
